mapping/index.go

```diff
@@ -447,6 +448,21 @@ func (im *IndexMappingImpl) FieldAnalyzer(field string) string {
 	return im.AnalyzerNameForPath(field)
 }
 
+// FieldMappingForPath returns the mapping for a specific field 'path'.
+func (im *IndexMappingImpl) FieldMappingForPath(path string) *FieldMapping {
+	for _, v := range im.TypeMapping {
+		for field, property := range v.Properties {
+			for _, v1 := range property.Fields {
+				if field == path {
+					// Return field mapping if the name matches the path param.
+					return v1
+				}
+			}
+		}
+	}
```

Comment:
From @Thejas-bhat : i think you might need to check the default mapping for the path as well (if the type mappings don't have the path in it). just like how AnalyzerNameForPath(paht string) does

Reply:
Done, pls take a look and let me know.